### ISSUE TITLE
[stable/nginx-ingress] allow nodePort for tcp/udp services

### ISSUE THREAD

**Issue by stephenmuss** [helm/charts#11917]
This is a duplicate of #8308 which was closed with lifecycle/stale.

**FEATURE REQUEST:** Support node ports for tcp/udp services.

**Which chart**: stable/nginx-ingress

Currently in the nginx-ingress chart it is only possible to specify a node port for http and https. Any ports specified using `tcp` and `udp` are automatically assigned a random node port. This is unsuitable when specifying `NodePort` for the service type.

I would be happy to submit a PR for this but would appreciate guidance on the preferred approach.

**Comment by stephenmuss** [comment]
I'm still happy to look at this but it would be good to get some input from the maintainers of the chart.

**Comment by stephenmuss** [comment]
Bump.

**Comment by ChiefAlexander** [comment]
A PR has been opened for this I believe: https://github.com/helm/charts/pull/13627

**Comment by ChiefAlexander** [comment]
/close
Fixed by https://github.com/helm/charts/pull/14171

**Comment by stephenmuss** [comment]
@ChiefAlexander 

Thanks for the update.

#14171 doesn't solve this from my perspective. That change accommodates TCP and UCP services as hostPorts only in daemonsets.

I'd like to see allowing nodePorts for the controller when running as a NodePort service type.

**Comment by ChiefAlexander** [comment]
Ah I see. Feel free to open up a PR that can accommodate that, I can review it once it is ready.

**Comment by stephenmuss** [comment]
Bump. Missed the last comment about a PR. Will see what I can do.
